### PR TITLE
tests: new core config helper

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This new helper is used to simplify how the core tests are configured.

Many tests have been modified to use the new helper

Also some repeated files have been removed and many small fixed have
been added.

This change also prepare the tests to support uc20.

Following PR including the update for tests on main suite is comming after this one.

### CODE REVIEW

File: tests/lib/core-config.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+}
+
+prepare_model(){
+    mv /var/lib/snapd/seed/assertions/model model.bak
+}
+
+restore_model(){
+    cp model.bak /var/lib/snapd/seed/assertions/model
```
Comment (zyga) [inline tests/lib/core-config.sh:17]: Why `cp` and not move? If this runs without a prior call to `prepare_model` then the stale data will be used.

File: tests/lib/core-config.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+prepare_seed(){
+    cp -f /var/lib/snapd/seed/seed.yaml seed.yaml.bak
```
Comment (zyga) [inline tests/lib/core-config.sh:21]: Why do we need `--force` here?

Comment (anonymouse64) [reply tests/lib/core-config.sh:21]: I assume because on uc20 this file (seed.yaml) doesn't exist

Comment (zyga) [reply tests/lib/core-config.sh:21]: I don't get it, this is what `--force` is documented to do:
```
       -f, --force
              if an existing destination file cannot be opened, remove it  and
              try  again  (this  option  is ignored when the -n option is also
              used)
```

File: tests/lib/core-config.sh (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/core-config.sh:17]: done, thanks for the review

File: tests/lib/core-config.sh (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/core-config.sh:21]: I updated how the seed is prepared. Thanks for your review

File: tests/lib/core-config.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    MODEL_FINAL=$(get_model "$MODEL_NAME")
+    cp -f "${TESTSLIB}/assertions/${MODEL_FINAL}" "/var/lib/snapd/seed/assertions/${MODEL_NAME}.model"
+}
+
+restore_test_model(){
+    MODEL_NAME=$1
+    rm -f "/var/lib/snapd/seed/assertions/${MODEL_NAME}.model"
+}
+
+prepare_pc_snap(){
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/core-config.sh:63]: Makes totally sense, thanks for the review.

File: tests/lib/core-config.sh [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    unsquashfs -no-progress /var/lib/snapd/snaps/pc_*.snap
+}
+
+complete_pc_snap(){
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/core-config.sh:67]: done

File: tests/lib/manip_seed.py [hunk 5]
```diff
@@ -18,11 +18,12 @@
         break
     i += 1
 
-# test-snapd-with-configure_123.snap -> test-snapd-configure
-snapname = path.basename(sys.argv[2]).split("_")[0]
-snaps.append(
-    {"name": snapname, "channel": "edge", "file": sys.argv[2],}
-)
+for snap in sys.argv[2:]:
+    # test-snapd-with-configure_123.snap -> test-snapd-configure
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/manip_seed.py:22]: Thanns for the review. We do that becausse snap pack produces this kind of output: test-snapd-tools_1.0_all.snap

File: tests/lib/core-config.sh [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+get_snap_suffix(){
+    if is_core18_system; then
+        echo "-core18"
+    elif is_core20_system; then
+        echo "-core20"
+    fi
+}
+
+wait_for_first_boot(){
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/core-config.sh]: done

File: tests/lib/core-config.sh [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+prepare_testrootorg_store(){
+    cp -f "$TESTSLIB"/assertions/testrootorg-store.account-key /var/lib/snapd/seed/assertions
+}
+
+restore_testrootorg_store(){
+    rm -f /var/lib/snapd/seed/assertions/testrootorg-store.account-key
+}
+
+prepare_test_account(){
+    ACCOUNT_NAME=$1
```
Comment (zyga) [inline tests/lib/core-config.sh:38]:
```suggestion
    local ACCOUNT_NAME=$1
```

File: tests/lib/core-config.sh [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    cp -f "$TESTSLIB/assertions/${ACCOUNT_NAME}.account" /var/lib/snapd/seed/assertions
+    cp -f "$TESTSLIB/assertions/${ACCOUNT_NAME}.account-key" /var/lib/snapd/seed/assertions
+}
+
+restore_test_account(){
+    ACCOUNT_NAME=$1
```
Comment (zyga) [inline tests/lib/core-config.sh:44]:
```suggestion
    local ACCOUNT_NAME=$1
```

File: tests/lib/core-config.sh [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    rm -f "/var/lib/snapd/seed/assertions/${ACCOUNT_NAME}.account"
+    rm -f "/var/lib/snapd/seed/assertions/${ACCOUNT_NAME}.account-key"
+}
+
+prepare_test_model(){
+    MODEL_NAME=$1
```
Comment (zyga) [inline tests/lib/core-config.sh:50]:
```suggestion
    local MODEL_NAME=$1
```

File: tests/lib/core-config.sh [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    MODEL_FINAL=$(get_model "$MODEL_NAME")
```
Comment (zyga) [inline tests/lib/core-config.sh:51]:
```suggestion
    local MODEL_FINAL=$(get_model "$MODEL_NAME")
```

File: tests/lib/core-config.sh [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    cp -f "${TESTSLIB}/assertions/${MODEL_FINAL}" "/var/lib/snapd/seed/assertions/${MODEL_NAME}.model"
+}
+
+restore_test_model(){
+    MODEL_NAME=$1
```
Comment (zyga) [inline tests/lib/core-config.sh:56]:
```suggestion
    local MODEL_NAME=$1
```

File: tests/lib/core-config.sh [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+       rm -f /etc/systemd/system/snap-pc-x1.mount
+       rm -f /etc/systemd/system/multi-user.target.wants/snap-pc-x1.mount
+       rm -f /var/lib/snapd/snaps/pc_x1.snap
+       systemctl daemon-reload
+    fi
+    rm -f /var/lib/snapd/seed/snaps/pc_x1.snap
+}
+
+get_model(){
+    MODEL_NAME=$1
```
Comment (zyga) [inline tests/lib/core-config.sh:82]:
```suggestion
    local MODEL_NAME=$1
```

File: tests/lib/manip_seed.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (zyga) [reply tests/lib/manip_seed.py:22]: I don't see how that is related to @cmatsuoka's comment? I also think that line should read
```
     # test-snapd-with-configure_123.snap -> test-snapd-with-configure
```

File: tests/lib/core-config.sh [hunk 13]
```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
```
Comment (zyga) [inline tests/lib/core-config.sh:1]: Is everything in this file a part a public function? Can you please add a comment about what each function is expected to do please.

I'm somewhat wary of another shell module that is sourced as those tend to be more fragile than executed programs. For instance, the interpreter here is "bash" but that's just a comment that is entirely ignored when this file is sourced from a non-bash shell program.

I'm also somewhat worried about naming, all the functions have free-form names. Shell does not issue any warning when a function is re-defined. I think we should either use a program that lives in `tests/lib/tools` and is invoked by path OR have a well-defined naming scheme as debugging a failure like that will not be fun.

Comment (sergiocazzolato) [reply tests/lib/core-config.sh:1]: @zyga, Agree regarding to the function names, I'll add function comments and update names

**Comment by sergiocazzolato** [comment]
@stolowski thanks

File: tests/core/snap-set-core-config/task.yaml [hunk 14]
```diff
@@ -89,6 +89,8 @@ execute: |
         exit 1
     fi
 
+
+    rm -f /var/lib/console-conf/complete
```
Comment (anonymouse64) [inline tests/core/snap-set-core-config/task.yaml:93]: could you explain this change? it seems unrelated to the rest of the PR?
